.changeset/fair-emus-divide.md

```diff
@@ -0,0 +1,39 @@
+---
+'astro': major
+'@astrojs/netlify': minor
+---
+
+The configuration `build.split` and `build.excludeMiddleware` are deprecated.
+
+Configuration that were inside the astro configuration, are now moved inside the adapter:
+
+```diff
+import {defineConfig} from "astro/config";
+import netlify from "@astrojs/netlify/functions";
+
+export default defineConfig({
+-    build: {
+-        excludeMiddleware: true
+-    },
+-    adapter: netlify()
++    adapter: netlify({
++        edgeMiddleware: true
++    })
+})
+```
+
+```diff
+import {defineConfig} from "astro/config";
+import vercel from "@astrojs/vercel/serverless";
+
+export default defineConfig({
+-    build: {
+-        split: true
+-    },
+-    adapter: netlify()
++    adapter: netlify({
++        functionPerRoute: true
++    })
+})
+```
```

Comment:
I think we only need one diff per changeset for netlify and vercel? Looks like it's duplicated here.